src/stats.ml

```diff
@@ -5,9 +5,28 @@ let enabled = ref false
 module Fd_count = struct
   type t = Unknown | This of int
 
+  let try_to_use_lsof () =
+    let temp = Filename.temp_file "dune" ".lsof" in
+    let stdout =
+      Unix.openfile temp [O_WRONLY; O_CREAT; O_TRUNC; O_SHARE_DELETE] 0o666
+    in
+    let prog = "/usr/sbin/lsof" in
+    let argv = [prog; "-w"; "-p"; string_of_int (Unix.getpid ())] in
+    let pid = Spawn.spawn ~prog ~argv ~stdout () in
+    Unix.close stdout;
+    match Unix.waitpid [] pid with
+    | _, Unix.WEXITED 0 ->
+      let num_lines = List.length (Io.input_lines (open_in temp)) in
+      This (num_lines - 1) (* the output contains a header line *)
+    | _ -> Unknown
```

Comment:
Why not use `Process.run_capture` (or one of its variations)?

Reply:
It was my first move but, if I remember correctly, it unfortunately
created a circular dependency.

Reply:
Indeed, comment added.